xwayland/xwm.c

```diff
@@ -1312,6 +1312,11 @@ static int x11_event_handler(int fd, uint32_t mask, void *data) {
 	xcb_generic_event_t *event;
 	struct wlr_xwm *xwm = data;
 
+	if ((mask & WL_EVENT_HANGUP) || (mask & WL_EVENT_ERROR)) {
+		xwm_destroy(xwm);
```

Comment:
`wlr_xwayland.xwm` will still be set to the free'ed memory after this. I think we need to add a destroy event to `wlr_xwm` to let `wlr_xwayland` do its cleanup, wdyt?

Reply:
Actually no need for this complexity. We have a `wlr_xwayland` field in `wlr_xwm`, we can just reset the pointer to NULL there.